core/src/util/usLog_p.h

```diff
@@ -69,25 +69,82 @@ struct LogMsg {
   std::stringstream buffer;
 };
 
+class Logger {
+public:
+    static Logger& instance()
+    { 
+      /* 
+      IMPORTANT: This is only thread safe if compiling with a C++11 compiler
+        which implements C++11 standard section 6.7.4:
+       "If control enters the declaration concurrently while the variable is being initialized, 
+        the concurrent execution shall wait for completion of the initialization."
+
+        However, even if a C++11 supported compiler isn't used, thread safe initialization
```

Comment:
On a second thought, having a static Logger instance is in conflict with the ability to create multiple `Framework` instances. So this needs a redesign anyway, possibly relying on a `ModuleContext` instance. I suggest to keep the code as is and add yet another issue to tackle this.